comp/dogstatsd/listeners/uds_common.go

```diff
@@ -299,23 +304,40 @@ func (l *UDSListener) handleConnection(conn *net.UnixConn) error {
 
 			log.Errorf("dogstatsd-uds: error reading packet: %v", err)
 			udsPacketReadingErrors.Add(1)
-			tlmUDSPackets.Inc(l.transport, "error")
+			tlmUDSPackets.Inc(listenerID, l.transport, "error")
 			continue
 		}
-		tlmUDSPackets.Inc(l.transport, "ok")
+		tlmUDSPackets.Inc(listenerID, l.transport, "ok")
 
 		udsBytes.Add(int64(n))
-		tlmUDSPacketsBytes.Add(float64(n), l.transport)
+		tlmUDSPacketsBytes.Add(float64(n), listenerID, l.transport)
 		packet.Contents = packet.Buffer[:n]
 		packet.Source = packets.UDS
 
 		// packetsBuffer handles the forwarding of the packets to the dogstatsd server intake channel
-		l.packetsBuffer.Append(packet)
+		packetsBuffer.Append(packet)
+	}
+}
+
+func getListenerID(conn *net.UnixConn) string {
+	f, err := conn.File()
+	if err != nil {
+		log.Errorf("dogstatsd-uds: error getting file from connection: %s", err)
 	}
+	return "uds-" + conn.LocalAddr().Network() + "-" + strconv.Itoa(int(f.Fd()))
```

Comment:
I'm concerned by `f.Fd()` for Agents collecting logs, where the file descriptor ID could grow quickly.
I know it's a pain for no additional technical win, but would it be hard to implement a set of used values (stored in the UDSListener instance), where the disconnected ones would remove themselves from, avoiding to be tight with the FD counts of the Agent process? I guess it could be if we let some connections die on their own without entering a `Stop()` or similar method? In current implementation it looks like it could be part of `clearTelemetry()`

Reply:
So I added a flag to enable per-socket telemetry, but decided to keep the file descriptor as suffix since this has a lot of value to debugging. Now that this is optional it should not be an issue anymore.